_inc/lib/debugger/class-jetpack-debug-data.php

```diff
@@ -0,0 +1,349 @@
+	 *
+	 * @return array $args Debug information in the same format as the initial argument.
+	 */
+	public static function core_debug_data( $debug ) {
+		$jetpack = array(
+			'jetpack' => array(
+				'label'       => __( 'Jetpack', 'jetpack' ),
+				'description' => sprintf(
+					/* translators: URL is to jetpack.com's contact support page. */
+					__( 'Diagnostic information helpful to <a href="%s" target="_blank" rel="noopener noreferrer">your Jetpack Happiness team</a>', 'jetpack' ),
+					esc_html( 'https://jetpack.com/contact-support' )
+				),
+				'fields'      => self::debug_data(),
+			),
+		);
+		$debug   = array_merge( $debug, $jetpack );
+		return $debug;
+	}
+
+	/**
+	 * Compile and return array of debug information.
+	 *
+	 * @since 7.3.0
+	 *
+	 * @return array $args {
+	 *          Associated array of arrays with the following.
+	 *         @type string  $label    The label for this piece of information.
+	 *         @type string  $value    The output that is of interest for this field.
+	 *         @type boolean $private  Optional. Set to true if data is sensitive (API keys, etc).
+	 * }
+	 */
+	public static function debug_data() {
+		$debug_info = array();
+
+		/* Add various important Jetpack options */
+		$debug_info['site_id']        = array(
+			'label'   => 'Jetpack Site ID',
+			'value'   => Jetpack_Options::get_option( 'id' ),
+			'private' => false,
+		);
+		$debug_info['ssl_cert']       = array(
+			'label'   => 'Jetpack SSL Verfication Bypass',
+			'value'   => ( Jetpack_Options::get_option( 'fallback_no_verify_ssl_certs' ) ) ? 'Yes' : 'No',
+			'private' => false,
+		);
+		$debug_info['time_diff']      = array(
+			'label'   => "Offset between Jetpack server's time and this server's time.",
+			'value'   => Jetpack_Options::get_option( 'time_diff' ),
+			'private' => false,
+		);
+		$debug_info['version_option'] = array(
+			'label'   => 'Current Jetpack Version Option',
+			'value'   => Jetpack_Options::get_option( 'version' ),
+			'private' => false,
+		);
+		$debug_info['old_version']    = array(
+			'label'   => 'Previous Jetpack Version',
+			'value'   => Jetpack_Options::get_option( 'old_version' ),
+			'private' => false,
+		);
+		$debug_info['public']         = array(
+			'label'   => 'Jetpack Site Public',
+			'value'   => ( Jetpack_Options::get_option( 'public' ) ) ? 'Public' : 'Private',
+			'private' => false,
+		);
+		$debug_info['master_user']    = array(
+			'label'   => 'Jetpack Master User',
+			'value'   => Jetpack_Options::get_option( 'master_user' ),
+			'private' => false,
+		);
+
+		/* Token information is private, but awareness if there one is set is helpful. */
+		$user_id     = get_current_user_id();
+		$user_tokens = Jetpack_Options::get_option( 'user_tokens' );
+		$blog_token  = Jetpack_Options::get_option( 'blog_token' );
+		if ( is_array( $user_tokens ) && array_key_exists( $user_id, $user_tokens ) ) {
+			$user_token = $user_tokens[ $user_id ];
+		} else {
+			$user_token = null;
```

Comment:
the `else` case can be omitted by setting default `$user_token` value before the `if` block

Reply:
Added via bbf5912c9